common/src/Preferences.cpp

```diff
@@ -223,6 +223,7 @@ Preference<float> CameraPanSpeed("Controls/Camera/Pan speed", 0.5f);
 Preference<bool> CameraPanInvertH("Controls/Camera/Invert horizontal pan", false);
 Preference<bool> CameraPanInvertV("Controls/Camera/Invert vertical pan", false);
 Preference<bool> CameraMouseWheelInvert("Controls/Camera/Invert mouse wheel", false);
+Preference<bool> CameraTrackpadMode("Controls/Camera/Trackpad mode", false);
```

Comment:
I was wondering why it is necessary to introduce a new preference for this. Zoom and rotate events will always come from the trackpad, and scroll events seem to have their source set to `Qt::MouseEventSynthesizedBySystem`, at least on macOS.

So if we know that an event came from the trackpad, would that be enough to avoid introducing this preference?

Reply:
Besides enabling support for the gestures, this preference also changes the behavior of the horizontal and vertical scroll events: With it disabled, vertical scrolling is still used for panning backward/forward (and zooming), so the workflow of existing users with mice is unaffected. With it enabled, scrolling horizontally and vertically are changed to behave as horizontal and vertical pans.

<img width="789" alt="image" src="https://github.com/user-attachments/assets/42050c40-e2cf-4614-8fcc-52a50824b81b">

Blender has a similar toggle that does roughly the same behavior change, which is where I copied the idea from.

Ideally we would be able to know if the scroll event came from a mouse or trackpad, and switch automatically, but `QWheelEvent` doesn't seem to expose that information, unfortunately. (And `QNativeGestureEvent` is not fired for two-finger scroll gestures)

Reply:
> Ideally we would be able to know if the scroll event came from a mouse or trackpad, and switch automatically, but `QWheelEvent` doesn't seem to expose that information, unfortunately. (And `QNativeGestureEvent` is not fired for two-finger scroll gestures)

But we do know: Zoom and rotate events will always come from the trackpad, and scroll events seem to have their source set to `Qt::MouseEventSynthesizedBySystem` (when they originate at the trackpad), at least on macOS.

Reply:
Is it possible to get a `QMouseEvent` from a `QWheelEvent`? The docs only list the `source()` getter for `QMouseEvent`

Reply:
I think you missed it, it's here: https://doc.qt.io/qt-5/qwheelevent.html#source

Reply:
Oh, I was looking at the _wrong_ `QWheelEvent`: https://doc.qt.io/qt-5/qt3dinput-qwheelevent.html 🥲

Looks like there's a different one inside of `Qt3DInput`